File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int leftMost(int[] nums,int target){
        int index=-1,mid;
        int start=0,end=nums.length-1;
        while(start<=end){
            mid=(start+end)/2;
            if(nums[mid]==target){
                index=mid;
                end=mid-1;
            }
            else if(nums[mid]<target){
                start=mid+1;
            }
            else{
                end=mid-1;
            } 
        }
        return index;
    }
    public int rightMost(int[] nums,int target){
        int index=-1,mid;
        int start=0,end=nums.length-1;
        while(start<=end){
            mid=(start+end)/2;
            if(nums[mid]==target){
                index=mid;
                start=mid+1;
            }
            else if(nums[mid]<target){
                start=mid+1;
            }
            else{
                end=mid-1;
            } 
        }
        return index;
    }
    public int[] searchRange(int[] nums, int target) {
        int[] result=new int[2];
        result[0]=leftMost(nums,target);
        result[1]=rightMost(nums,target);
        return result;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def leftMost(self,nums,target):
        index=-1
        start,end=0,len(nums)-1
        while start<=end:
            mid=(start+end)//2
            if nums[mid]==target:
                index=mid
                end=mid-1
            elif nums[mid]<target:
                start=mid+1
            else:
                end=mid-1
        return index
    def rightMost(self,nums,target):
        index=-1
        start,end=0,len(nums)-1
        while start<=end:
            mid=(start+end)//2
            if nums[mid]==target:
                index=mid
                start=mid+1
            elif nums[mid]<target:
                start=mid+1
            else:
                end=mid-1
        return index

    def searchRange(self, nums: List[int], target: int) -> List[int]:
        result=[0,0]
        result[0]=self.leftMost(nums,target)
        result[1]=self.rightMost(nums,target)
        return result